src/concepts/kafka-proxies/http-proxy.md

```diff
@@ -0,0 +1,67 @@
+
+## Correlated Request-Response
+
+Zilla manages the HTTP lifecycle with the request and response payload on the event stream over a pair of Kafka topics. Each request message is correlated to the corresponding response message with a `zilla:correlation-id` header, providing an identifier for both Zilla and Kafka workflows to act on.
+
+### sync
+
+A synchronous interaction from the client will open a connection, producing a request message. The connection will remain open, waiting for the correlated response message to be fetched and returned to the caller.
+
+### async
+
+An asynchronous interaction includes a `prefer: respond-async` header. After producing a request message, the connection will immediately return with `202 Accepted` plus the location path to retrieve a correlated response. The client then sends a request to the returned location path with the `prefer: wait=N` header to retrieve the correlated response as soon as it becomes available, removing the need for client polling.
+
+## SSE Streaming
+
+The Zilla Server-sent Events (SSE) Kafka Proxy exposes an SSE stream of Kafka messages using the [sse-kafka](../../reference/config/bindings/binding-sse-kafka.md) binding.
+
+An [SSE](https://html.spec.whatwg.org/multipage/server-sent-events.html) server allows a web browser using the `EventSource` interface to open a connection and receive a stream of text from the server, interpreted as individual messages. Zilla relays text messages on a Kafka topic into the event stream.
+
+### Message Filtering
+
+The message source topic is defined in a route, and the route is matched by the path defined for the client to connect. A route can [filter](../../reference/config/bindings/binding-sse-kafka.md#routes-with) the messages delivered to the SSE stream using the message key and headers. A filter's value can be statically defined in the config or be pulled from a [path param](../../concepts/config-intro.md#dynamic-path-parameters).
+
+### Reliable Delivery
+
+Zilla sends the event id and last-event-id header to recover from an interrupted stream without message loss and without needing the client to acknowledge message receipt explicitly. An interrupted SSE stream is recovered by connecting to any Zilla instance in the same auto-scaling group because each Zilla instance is stateless.
+
+## Oneway
+
+Clients can produce fire and forget HTTP request payload to a Kafka topic. The Kafka message key and headers are set using [path params](../../concepts/config-intro.md#dynamic-path-parameters).
+
+## Caching
+
+Bindings can retrieve messages from a Kafka topic, filtered by message key and headers, with the key and header values extracted from the [path params](../../concepts/config-intro.md#dynamic-path-parameters).
+
+An HTTP response returns with an `etag` header. This fetch supports a conditional `GET if-none-match request` returning `304` if not modified or `200` if modified (with a new `etag` header).
```

Comment:
We also support prefer:wait=N with cache control no cache and if none match etag to wait for updated message.